### PR TITLE
Updating declarations of openpgpjs and adding sign and verify

### PR DESCRIPTION
**Pull request by ecamellini**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/openpgpjs/openpgpjs/blob/v4.0.1/src/openpgp.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by errietta — CHANGES_REQUESTED** [review]
Thanks! Can you please look at updating the types for ts3.2 and later as well in the ts3.2/ directory?
When I get some time I'll try doing this with a common directory as well, as right now there's some duplication here thanks to the different syntax for one of the types in >= 3.2 vs < 3.2  😞

File: types/openpgp/index.d.ts [hunk 1]
```diff
@@ -39,6 +40,8 @@ export interface EncryptOptions {
 export interface EncryptedMessage {
     data?: string,
     message?: message.Message,
+    signature?: String |ReadableStream<String> |Signature // TODO add NodeStream
```
Comment (errietta) [inline types/openpgp/index.d.ts]: This causes the build failure as well, you can have `ReadableStream<String>` in `ts3.2/` but not in ts3.1, so you can make this change in the `ts3.2` dir and in this file make it just `ReadableStream`

**Review by errietta — CHANGES_REQUESTED** [review]

File: types/openpgp/index.d.ts (hunk 1 above)
Comment (ecamellini) [reply types/openpgp/index.d.ts]: @errietta done 👍

File: types/openpgp/index.d.ts [hunk 2]
```diff
@@ -39,6 +40,8 @@ export interface EncryptOptions {
 export interface EncryptedMessage {
     data?: string,
     message?: message.Message,
+    signature?: string | ReadableStream | Signature // TODO add NodeStream
```
Comment (errietta) [inline types/openpgp/index.d.ts]: Did you mean to do those todos now or later?

**Comment by errietta** [comment]
Btw, if you want, you can also put your name in `index.d.ts` with the contributors 😄

**Review by errietta — APPROVED** [review]

File: types/openpgp/index.d.ts (hunk 2 above)
Comment (ecamellini) [reply types/openpgp/index.d.ts]: We could do leave them to be done in a future PR, if you agree!

**Review by errietta — APPROVED** [review]

**Pull request merged**